mlx.c

```diff
@@ -28,32 +38,47 @@ int	clean_exit(t_struct *mlx)
 	exit(0);
 }
 
-int	event_handler(int keycode, t_struct *mlx)
+int	key_down(int keycode, t_struct *mlx)
 {
 	if (keycode == KEY_ESC)
 		clean_exit(mlx);
-	if (keycode == KEY_A || keycode == KEY_D
-		|| keycode == KEY_S || keycode == KEY_W)
-		movement(keycode, mlx);
-	if (keycode == KEY_LEFT || keycode == KEY_RIGHT)
-		rotate(keycode, mlx);
-	clear_screen(mlx);
-	ray_cast(mlx);
-	mlx_put_image_to_window(mlx->mlx, mlx->mlx_win, mlx->img, 0, 0);
+	if (keycode == KEY_W)
+		mlx->key.forward = true;
+	if (keycode == KEY_S)
+		mlx->key.backward = true;
+	if (keycode == KEY_A)
+		mlx->key.left = true;
+	if (keycode == KEY_D)
+		mlx->key.right = true;
+	if (keycode == KEY_LEFT)
+		mlx->key.r_left = true;
+	if (keycode == KEY_RIGHT)
+		mlx->key.r_right = true;
 	return (0);
 }
 
-int	mouse_move(int x, int y, t_struct *mlx)
+int	key_up(int keycode, t_struct *mlx)
 {
-	if (x < mlx->prev_x)
-		rotate(KEY_LEFT, mlx);
-	else
-		rotate(KEY_RIGHT, mlx);
-	mlx->prev_x = x;
-	clear_screen(mlx);
-	ray_cast(mlx);
-	mlx_put_image_to_window(mlx->mlx, mlx->mlx_win, mlx->img, 0, 0);
-	return (y - y);
+	if (keycode == KEY_W || keycode == KEY_S || keycode == KEY_A
+		|| keycode == KEY_D)
+	{
+		if (keycode == KEY_W)
+			mlx->key.forward = False;
```

Comment:
why in previous function you have boolean called from lower case and here from upper?

Reply:
I forgot to change it to lowercase. The camel case ```True``` or ```False``` is extended from the mlx library. Should not effect the behaviour of the code.